mypy / error: Missing named argument ... [call-arg] / for optional kwargs

* bleak version: 0.21.1
* Python version: 3.10, 3.12
* Operating System: Ubuntu 22.04, Windows 11
* BlueZ version (`bluetoothctl -v`) in case of Linux: NA

### Description

Mypy treats kwargs that are `Unpack`ed as required.  In fact, these kwargs were always intended as platform-specific optional arguments.  

Mypy will diagnose missing kwargs as errors.  In bleak, this occurs for any function that types kwargs as `Unpack[ExtraArgs]`.  These include the methods discover, find_device_by_address, find_device_by_name, and find_device_by_filter.

Where a more friendly solution may be to use `Generic` and type aliases, use of `Unpack` is a welcome improvement over kwargs, so I believe that it's simple enough to add total=False because ALL of the kwargs are optional.  Refer to https://peps.python.org/pep-0692/#required-and-non-required-keys.  If finer grained control is required, then I suggest Generic and Type Aliases so that the user can see the arguments in their editor instead of having to rely on documentation.

### What I Did

Ran mypy 1.7.1 on a file like:

```python
from bleak import BleakScanner

async def main() -> None:
    await BleakScanner.find_device_by_address("address")
```

`mypy .` will diagnose every missing kwarg as error:

```
smpclient\blah.py:5: error: Missing named argument "service_uuids" for "find_device_by_address" of "BleakScanner"  [call-arg]
smpclient\blah.py:5: error: Missing named argument "scanning_mode" for "find_device_by_address" of "BleakScanner"  [call-arg]
smpclient\blah.py:5: error: Missing named argument "bluez" for "find_device_by_address" of "BleakScanner"  [call-arg]
smpclient\blah.py:5: error: Missing named argument "cb" for "find_device_by_address" of "BleakScanner"  [call-arg]
smpclient\blah.py:5: error: Missing named argument "backend" for "find_device_by_address" of "BleakScanner"  [call-arg]
```


### Logs

N/A

--- a/AUTHORS.rst
+++ b/AUTHORS.rst
@@ -22,6 +22,7 @@ Contributors
 * Edward Betts <edward@4angle.com>
 * Robbe Gaeremynck <robbe.gaeremynck@outlook.com>
 * David Johansen <davejohansen@gmail.com>
+* JP Hutchins <jphutchins@gmail.com>
 
 Sponsors
 --------

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -22,6 +22,7 @@ Changed
 Fixed
 -----
 * Fixed BlueZ version in passive scanning error message. Fixes #1433.
+* Fixed mypy requiring ``Unpack[ExtraArgs]`` that were intended to be optional.  Fixes #1487.
 
 `0.21.1`_ (2023-09-08)
 ======================

--- a/bleak/__init__.py
+++ b/bleak/__init__.py
@@ -245,7 +245,7 @@ class BleakScanner:
         finally:
             unregister_callback()
 
-    class ExtraArgs(TypedDict):
+    class ExtraArgs(TypedDict, total=False):
         """
         Keyword args from :class:`~bleak.BleakScanner` that can be passed to
         other convenience methods.

